pkg/shell/plot.js

```diff
@@ -163,54 +167,53 @@ shell.plot = function plot(element, x_range_seconds, x_stop_seconds) {
     }
 
     function start_walking() {
-        /* Don't overflow 32 signed bits with the interval.  This
-         * means that plots that would make about one step every month
-         * don't walk at all, but I guess that is ok.
-         */
-        if (interval > 2000000000)
-            return;
-
-        if (!walk_timer)
-            walk_timer = window.setInterval(function () {
-                now += interval;
-                for (var i = 0; i < data.length; i++)
-                    data[i].walk();
-                refresh();
-            }, interval);
+        if (grid)
+            grid.walk();
     }
 
     function stop_walking() {
-        if (walk_timer)
-            window.clearInterval(walk_timer);
-        walk_timer = null;
+        if (grid)
+            grid.move(grid.beg, grid.end);
     }
 
     function reset(x_range_seconds, x_stop_seconds) {
         if (flot)
             flot.clearSelection(true);
 
-        now = (new Date()).getTime();
+        if (x_range_seconds !== undefined) {
 
-        if (x_stop_seconds !== undefined)
-            x_offset = now - x_stop_seconds*1000;
-        else
-            x_offset = 0;
+            // Fill the plot with about 1000 samples, but don't sample
+            // faster than once per second.
+            //
+            // TODO - do this based on the actual size of the plot.
```

Comment:
I want to use a different interval when the range changes, so that we don't sample one week of data with an interval of 1 second.